kfs-core/src/main/java/edu/arizona/kfs/vnd/service/impl/EcustomsServiceImpl.java

```diff
@@ -0,0 +1,698 @@
+        if (lookupService == null) {
+            lookupService = SpringContext.getBean(LookupService.class);
+        }
+        return lookupService;
+    }
+
+    private static EcustomsFileService getEcustomsFileService() {
+        if (ecustomsFileService == null) {
+            ecustomsFileService = SpringContext.getBean(EcustomsFileService.class);
+        }
+        return ecustomsFileService;
+    }
+
+    @Override
+    public boolean createEcustomsDailyFile(String jobName, Date jobRunDate) throws Exception {
+        LOG.debug("createEcustomsDailyFile(): jobRunDate=" + jobRunDate + ", start-time=" + new Date());
+        boolean processVendorDetails = false;
+        try {
+            boolean deleteOldDailyDone = getEcustomsFileService().deleteOldDailyDoneFiles();
+            LOG.debug("getEcustomsFileService().deleteOldDailyDoneFiles() returned " + deleteOldDailyDone);
+            dataFile = getEcustomsFileService().getDailyBatchDataFile(jobRunDate);
+            vendorCountFile = getEcustomsFileService().getDailyBatchVendorCountFile(jobRunDate);
+            batchName = VendorConstants.ECUSTOMS_BATCH_DAILY;
+            List<VendorDetail> vendorDetails = getDailyVendorDetailList(jobRunDate);
+            processVendorDetails = processVendorDetails(jobRunDate, vendorDetails);
+            getEcustomsFileService().createDailyBatchDoneFile(jobRunDate);
+        } catch (Exception e) {
+            LOG.error("Exception: " + e.getMessage());
+            throw new Exception(e.getMessage());
+        }
+        return processVendorDetails;
+    }
+
+    /**
+     * Processes the list of vendors for output.
+     *
+     * @param jobRunDate
+     *
+     * @param vendorDetails
+     * @return
+     * @throws IOException
+     */
+    private boolean processVendorDetails(Date jobRunDate, List<VendorDetail> vendorDetails) throws IOException {
+        FileOutputStream dataFileStream = new FileOutputStream(dataFile);
+        List<String> outputVendorLineList = new ArrayList<String>();
+//        List<VendorDetailExtension> updatedVDEList = new ArrayList<VendorDetailExtension>();
+//        List<Note> noteList = new ArrayList<Note>();
+
+        for (VendorDetail vendorDetail : vendorDetails) {
+            String vendorLine = generateVendorLine(vendorDetail);
+            if (!outputVendorLineList.contains(vendorLine)) {
+                outputVendorLineList.add(vendorLine);
+                dataFileStream.write(vendorLine.getBytes());
+//                VendorDetailExtension updatedVDE = updateVendorDetailExtension(vendorDetail);
+                updateVendor(vendorDetail);
+//                Note note = createNote(vendorDetail);
+//                updatedVDEList.add(updatedVDE);
+//                noteList.add(note);
+            }
+        }
+//        getBusinessObjectService().save(updatedVDEList);
+//        getBusinessObjectService().save(noteList);
+        dataFileStream.flush();
+        dataFileStream.close();

```

Comment:
dataFileStream needs to be in try/finally to ensure resource gets closed. Do try to minimize what's in all try's, since the compiler will not optimize contents therein.